frontend/src/pages/editProfile.js

```diff
@@ -1,40 +1,91 @@
-import React from "react";
-import { useSelector, useDispatch } from "react-redux";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 
-import { editProfile } from "../store/profile";
 import InputField from "../components/input";
-import { validate, validateField, getErrorPayload } from "../utils/validation";
+import { validate, validateField } from "../utils/validation";
+import * as profileService from "../services/profileService";
+import { getErrorMessage } from "../utils/validation";
 
 const EditProfile = () => {
-  const state = useSelector((state) => state.editProfile.value);
-  const dispatch = useDispatch();
+  const [state, setState] = useState({ errors: {} });
+
   const navigate = useNavigate();
+  let avatarFile = {};
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      const { data } = await profileService.getCurrentProfile();
+      const profile = data.profile;
+      delete profile["avatar"];
+      delete profile["total_words_learned"];
+      delete profile["total_lessons_learned"];
+      const user = data.user;
+      const errors = {};
+      const payload = { ...state, ...profile, ...user, errors };
+
+      setState(payload);
+    };
+    fetchProfile();
+  }, []);
+
+  const handleSubmit = async () => {
+    const currErrors = validate(state);
+    setState({ ...state, errors: currErrors });
+
+    if (Object.keys(currErrors).length === 0) {
+      try {
+        const profileResponse = await profileService.update(state);
 
-  const handleSubmit = () => {
-    const errors = validate(state);
-    const payload = { ...state, errors };
-    dispatch(editProfile(payload));
-    // navigate("/profile");
+        const avatarResponse = await profileService.uploadAvatar(
+          state,
+          avatarFile
+        );
+        if (avatarResponse.status === 201 && profileResponse.status === 201) {
+          // navigate("/profile");
+        }
+      } catch (exception) {
+        if (exception.response) {
+          // if failed, update state and show error
+          const response_error = exception.response.data;
+          const key = Object.keys(response_error);
+          const message = response_error[key];
+          const error = { [key]: message };
+
+          setState({ ...state, errors: error });
```

Comment:
Ano data structure ng response_error? maybe pwd dretso eset sa error state
```suggestion
          const error = exception.response.data;
          setErrors(error);
```